constraints_test.go

```diff
@@ -430,6 +436,27 @@ func TestBidirectionalSerialization(t *testing.T) {
 	}
 }
 
+func TestPreferUnaryOpForm(t *testing.T) {
+	tests := []struct {
+		in, out string
+	}{
+		{">=0.1.7, <0.2.0", "^0.1.7"}, // carat shifting below 1.0.0
+		{">=1.1.0, <2.0.0", "^1.1.0"},
+		{">=1.1.0, <2.0.0, !=1.2.3", "^1.1.0, !=1.2.3"},
```

Comment:
What is the motivation for the shorthand being the default? Curious of the thinking behind it.